File: labs/03/YearCodes.java
/* YearCodes.java is a driver for function yearCode()
 *
 * Input: the name of the student's year
 * Precondition: the name is a lowercase string value
 * Output: the number corresponding to that year name
 *
 * Begun by: Dr. Adams, for CS 214, at Calvin College
 * Completed by: Ian Christensen, for CS 214, at Calvin College
 * Date: February 14, 2018
 */

import java.util.Scanner; // import the java scanner for user input capability

public class YearCodes {  // declare the class for YearCodes

   public static void main(String[] args) { // declare the class main
      Scanner keyboard = new Scanner(System.in); // create a scanner object for input capability
      System.out.println("\nEnter your academic year: "); // prompt
      String year = keyboard.next();      // read year
      System.out.println(yearCode(year)); // display its code
   }  //main method

   /* Function: yearCode() computes a student's year integer, given an academic year
    * Parameter: year, a string
    * Precondition: year is one of {freshman, sophomore, junior, or senior}.
    * Returns: the integer code corresponding to year.
    */
   public static int yearCode(String year) {  // declare the function/method definition for yearCode
      if (year.equals("freshman"))  // check for freshman inputs
        return 1; // return the corresponding year for freshman
      else if (year.equals("sophomore")) // check for sophomore inputs
        return 2; // return the corresponding year for sophomores
      else if (year.equals("junior")) // check for junior inputs
        return 3; // return the corresponding year for juniors
      else if (year.equals("senior")) // check for senior inputs
        return 4; // return the corresponding year for seniors
      else  // catch any unexpected inputs
        return 0; // return the value zero for all invalid inputs
   }  // yearCode method
} // class YearCodes

File: labs/03/year_codes.rb
# !/usr/bin/ruby
# year_codes.rb translates an academic year into a numeric code
# Begun by: Dr. Nelesen, for CS 214 at Calvin College 
# Completed by: Ian Christensen, for CS 214, at Calvin College
# Date: February 13, 2018
#
# Input: The name of an academic year
# Precondition: The academic year is a string with a value of freshman, sophomore, junior or senior
# Output: The corresponding integer code for the given academic year

def yearCode year	# declare the yearCode function/method definition
	if year =~ /freshman/	# check for freshman
		1	# return the corresponding number
	elsif year =~ /sophomore/	# check for sophomore
		2	# return the corresponding number
	elsif year =~ /junior/	# check for junior
		3	# return the corresponding number
	elsif year =~ /senior/	# check for senior
		4	# return the corresponding number
	else	# catch any unexpected inputs
		0	# return 0 for all invalid inputs
	end	# end the if statment...this took me forever to figure out...
end	# end the yearCode function

if __FILE__ == $0	# if the file is being run
   print "Enter the year: "	# prompt the user for the year
   year = gets	# store the input in year
   print "Numeric code is: "	# output a message
   puts yearCode(year)	# output the corresponding year
end	# end the if statement

